pkg/controller/manager/manager_controller.go

```diff
@@ -205,6 +205,20 @@ func (r *ReconcileManager) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	// Check that if the manager certpair secret exists that it is valid (has key and cert fields)
+	// If it does not exist then this function returns a nil secret but no error and a self-signed
+	// certificate will be generated when rendering below.
+	tlsSecret, err := utils.ValidateCertPair(r.client,
+		render.ManagerTLSSecretName,
+		render.ManagerSecretKeyName,
+		render.ManagerSecretCertName,
+	)
+	if err != nil {
+		log.Error(err, "Invalid TLS Cert")
+		r.status.SetDegraded("Error validating TLS certificate", err.Error())
+		return reconcile.Result{}, err
+	}
+
```

Comment:
Also when installation.clusterManagementType indicates management then we should validate that tlsSecret is not nil.

Reply:
Added in https://github.com/tigera/operator/pull/541/commits/2ca889a930515473ddc193328e46b06c8760bc73